src/libbson/src/bson/bson.c

```diff
@@ -3584,7 +3584,19 @@ _bson_iter_validate_document (const bson_iter_t *iter,
       state->phase = BSON_VALIDATE_PHASE_LF_REF_KEY;
    }
 
-   (void) bson_iter_visit_all (&child, &bson_validate_funcs, state);
+   bson_iter_visit_all_v2 (
```

Comment:
Should the result of this call be part of the condition below? e.g.
```c
   if (!bson_iter_visit_all_v2 (
          &child, &bson_validate_funcs, BSON_ITER_VISIT_NOFLAGS, state) &&
       child.err_off > 0 && state->err_offset < 0) {
```

Reply:
By careful reading of the source, the return value from this function is nearly meaningless, because it can return `true` or `false` both for success and for failure. It's just quite unfortunate that we're trying to cram too much information into a single `bool`.